runtime/server/queries_metrics.go

```diff
@@ -401,6 +401,47 @@ func (s *Server) MetricsViewSearch(ctx context.Context, req *runtimev1.MetricsVi
 	return q.Result, nil
 }
 
+func (s *Server) MetricsViewResolveTimeRanges(ctx context.Context, req *runtimev1.MetricsViewResolveTimeRangesRequest) (*runtimev1.MetricsViewResolveTimeRangesResponse, error) {
+	observability.AddRequestAttributes(ctx,
+		attribute.String("args.instance_id", req.InstanceId),
+		attribute.String("args.metric_view", req.MetricsViewName),
+		attribute.StringSlice("args.rill_times", req.RillTimes),
+	)
+
+	if !auth.GetClaims(ctx).CanInstance(req.InstanceId, auth.ReadMetrics) {
+		return nil, ErrForbidden
+	}
+	s.addInstanceRequestAttributes(ctx, req.InstanceId)
+
+	mv, security, err := resolveMVAndSecurity(ctx, s.runtime, req.InstanceId, req.MetricsViewName)
+	if err != nil {
+		return nil, err
+	}
+
+	timeRangeQuery := &queries.MetricsViewTimeRange{
+		MetricsViewName:    req.MetricsViewName,
+		MetricsView:        mv.ValidSpec,
+		ResolvedMVSecurity: security,
+	}
+	err = s.runtime.Query(ctx, req.InstanceId, timeRangeQuery, int(req.Priority))
+	if err != nil {
+		return nil, err
+	}
+
+	q := &queries.MetricsViewResolveTimeRanges{
+		MetricsViewName: req.MetricsViewName,
+		MinTime:         timeRangeQuery.Result.TimeRangeSummary.Min.AsTime(),
```

Comment:
It's weird that it resolves `MinTime` separately here, but it resolves the watermark/max time inside the `MetricsViewResolveTimeRanges` implementation.

I would expect a) that it generally resolves these timestamps the same way in most places, b) that it resolves these timestamps at the same place / subject to the same caching.

Reply:
Will raise a follow up to use the metrics_time_range resolver.